src/Control/Controller.php

```diff
@@ -131,6 +138,10 @@ public function callback(HTTPRequest $request)
             $session->clear('oauth2');
             $this->httpError(400, 'Invalid session state.');
             return null;
+        } else if ($this->validateState($request) === 'isPost') {
+            return $this->renderWith('OAuthRedirect', [
+                'url' => sprintf("/%s?code=%s&state=%s", $request->getURL(), $request->postVar('code'), $request->postVar('state'))
```

Comment:
Isn’t this redirecting back to the same URL, just a GET instead of a POST? Why does this render a template that then redirects, instead of performing a redirect directly?

Reply:
Apple makes a POST instead of a GET request which leads to CORS problems. This intermediate step was introduced to make it a GET request.

$this->redirect() would not work because the request has to happen on the client side (via JavaScript).